### DISCUSSION TITLE
Why would rector not refactor mysql code?

### DISCUSSION THREAD

**Discussion by 7thstorm** [rectorphp/rector#8025] [Q&A]
I tried using the single levelset, as well as individual 4 rules pertaining to mysql. yet, each time, there were no changes. I tried the online demo and that too showed no changes

In my php file, It seems I have the matching lines of code so rector should have changed. 
what am I missing?

The lines of code in my php file:

```
$rs = mysql_query($qry);
 mysql_query($upsql) or die(mysql_error());
 $result = mysql_query($sql);
```

should it not have changed these lines??

**Comment by 7thstorm** [comment]
I also tried to run it on the demo site:

https://getrector.com/demo/cd21c81c-8650-474b-9360-e287b07a5372

but I keep getting this error:

```
Rector run Failed:

The process "'../vendor/bin/rector' 'process' '/var/www/src/../storage/demo/1y7hr8410495bpj2za0m/rector_analyzed_file.php' '--config' '/var/www/src/../storage/demo/1y7hr8410495bpj2za0m/rector.php' '--output-format' 'json'" exceeded the timeout of 60 seconds.
```

The above is using  SetList::MYSQL_TO_MYSQLI

I did further testing using the 4 mysql rules separately.

It gets hung up consistently on one rule:

https://getrector.com/demo/a84bd1a1-0b37-4a7f-bc4f-eb066949f627

` $rectorConfig->rule(\Rector\MysqlToMysqli\Rector\FuncCall\MysqlQueryMysqlErrorWithLinkRector::class);`

If I run the other three rules, it runs fine, but tells me no change

Here is the test link:

https://getrector.com/demo/26819039-9a98-4b81-a0c2-6a47d1e31664

Hope this is enough for someone to give me an idea about what's going on

**Reply by samsonasik** [reply]
`mysqli_query` require a connection instance as first argument, the `MysqlQueryMysqlErrorWithLinkRector` can't find it, It seems you define connection somewhere in include file, if connection defined, it will works as expected

https://getrector.com/demo/df886b8e-9d23-4f05-9469-44a6dfdb699d

You may need to skip this case.

**Reply by 7thstorm** [reply]
OK. understood. Maybe in the future, a change can be made so it skips this rule if connection is not defined, particularly, if running script o multiple files in a folder
what about the other three rules? why would it show OK, should it not have proposed changes?

https://getrector.com/demo/0cee47c5-2e3b-4912-b2d1-c99c3068fc21

**Reply by samsonasik** [reply]
Let's focus to resolve issue one by one, the first thing to do is ensure no infinite loop when no connection instance defined. 

Feel free to provide a patch first if you have a chance ;)

**Reply by samsonasik** [reply]
https://github.com/rectorphp/rector-src/pull/4375
